scripts/check-events

```diff
@@ -38,48 +58,72 @@ class Event
+    EVENT_CLASSES.each do |suffix, (required_message, counters, additional_tags)|
+      if @name.end_with? suffix
+        @logs.each do |type, message, parameters|
+          if type != 'trace'
+            @reports.append('Log type MUST be \"trace!\".')
+          end
+          if message != required_message
+            @reports.append("Log message MUST be \"#{required_message}\" (is \"#{message}\").")
+          end
+          additional_tags.each do |tag_name|
+            unless parameters.include? tag_name
+              @reports.append("Log MUST contain tag \"#{tag_name}\"")
+            end
+          end
+        end
+        counters.each do |counter|
+          counter = "component_#{counter}_total"
+          counters_must_include(counter, additional_tags - BYTE_SIZE_COUNT)
+        end
+      end
     end
 
     has_errors = @logs.one? { |type, _, _| type == 'error' }
 
     # Make sure Error events output an error
     if has_errors or @name.end_with? 'Error'
       append('Error events MUST be named "___Error".') unless @name.end_with? 'Error'
-      counters_must_include('component_errors_total', ['error_type', 'stage'] + @members - ['body', 'error', 'text'])
+      counters_must_include('component_errors_total', ['error_type', 'stage'])
     end
 
     # Make sure error events contain the right parameters
     @logs.each do |type, message, parameters|
       if type == 'error'
-        ['error', 'stage'].each do |parameter|
+        ['error_type', 'stage'].each do |parameter|
           unless parameters.include? parameter
             @reports.append("Error log MUST include parameter \"#{parameter}\".")
           end
         end
+
+        ['component_errors_total', 'component_discarded_events_total'].each do |event_name|
+          if @counters[event_name]
+            ['error_code', 'error_type', 'stage'].each do |parameter|
+              if parameters.include? parameter and !@counters[event_name].include? parameter
+                @reports.append("Counter \"#{event_name}\" must include \"#{parameter}\" to match error log.")
+              end
+            end
+          end
+        end
+      end
+    end
+
+    @counters.each do |name, tags|
+      # Only component_errors_total and component_discarded_events_total metrics are considered
+      if ['component_errors_total', 'component_discarded_events_total'].include? name
+        # Make sure defined tags to counters are constants
+        tags.each do |tag, value|
+          if tag == 'stage'
+            if !value.start_with? 'error_stage::'
+              @reports.append("Counter \"#{name}\" tag \"#{tag}\" value must be an \"error_stage\" constant.")
+            end
+          elsif tag == 'error_type'
+            if !value.start_with? 'error_type::'
+              @reports.append("Counter \"#{name}\" tag \"#{tag}\" value must be an \"error_type\" constant.")
+            end
+          end
```

Comment:
This appears to have dropped the check on `error_code` being constant. Is that not a useful check?

Reply:
As we concluded when discussing with @jszwedko, `error_code` might not always be constants. For example, when being build out of a http status code, we end up not having a constant but something like `format!("http_status_{}", status_code)`. That's why strictly forcing it to be a constant doesn't work.

Reply:
Ah, right, got it.